### ISSUE TITLE
Create well defined API for medical history questions and data

### ISSUE THREAD

**Issue by codeJRV** [GoCorona-org/Backend#18]

**Comment by theTechie** [comment]
is this API ready yet?

I see the form here - http://52.66.156.232:8000/corona_app/medmap/

But, irrespective of whether i provide some value or not. It's always returning 200. and the response is a HTML page.

Can we have a REST API where the response is a JSON?  Also, is the result (i.e. probability of having the virus) based on the questions answered returned immediately or it takes time to compute it?

**Comment by wadkar** [comment]
I can answer the immediate vs async part. Currently the django backend is *not* async, it will process your request and make you wait for response.

This, however, I don't think is scalable in the long run. We would like to use Django 3.0's ASGI interface and convert this interface into async. However, this has issue for the database connections because Django's ORM is not threadsafe so something like redis worker is required to manage the data processing.

Long story short - assume it's "immediate". If, and when, we change the backend to async we'll revisit this issue.

**Comment by theTechie** [comment]
> I can answer the immediate vs async part. Currently the django backend is _not_ async, it will process your request and make you wait for response.
> 
> This, however, I don't think is scalable in the long run. We would like to use Django 3.0's ASGI interface and convert this interface into async. However, this has issue for the database connections because Django's ORM is not threadsafe so something like redis worker is required to manage the data processing.
> 
> Long story short - assume it's "immediate". If, and when, we change the backend to async we'll revisit this issue.

Ok. thanks for the reply @wadkar What's the status on this API though? Doesn't look like its functioning as expected?

Also, i observed i am not able to do a simple POST call through a API client. Getting a CSRF verification error (403 - Forbidden). When are we going to RESTify these APIs ? these APIs are hard to consume with Django i believe?

![image](https://user-images.githubusercontent.com/1046486/80011108-02f3cd00-84e9-11ea-852d-3a26cfa88a87.png)

**Comment by sanjanamoha** [comment]
The async functionality is not required for medical data but for intersection calculation.
The result is found in the response of "/medmap/result/{med_uuid}" URL.
The HTML format was removed and JSON format has been factored in the API

The most crucial issue of API/UI mismatch has been shifted to:
https://github.com/GoCorona-org/Backend/issues/35
